libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/PrivateKeyStore.java

```diff
@@ -68,15 +68,31 @@ default String getApplicationSecret(String appName, String keyName) {
 
     /**
      * Retrieve the application secret based on the configured key name as char[].
+     * @deprecated
      * The application name specifies what component is this secret for;
      * for example, jdbc for accessing the secret for the jdbc user.
      * The default implementation assumes the key name is the secret.
      * @param appName application name for the secret
      * @param keyName configured value for the secret
      * @return secret for the given key and application as char[]
      */
+    @Deprecated
     default char[] getSecret(String appName, String keyName) {
         final String secret = getApplicationSecret(appName, keyName);
         return secret != null ? secret.toCharArray() : null;
     }
+
+    /**
+     * Retrieve the application secret based on the configured key name as char[].
+     * The application name specifies what component is this secret for;
+     * for example, jdbc for accessing the secret for the jdbc user.
+     * The default implementation assumes the key name is the secret.
+     * @param appName application name for the secret
+     * @param keygroupName key group name for the secret
+     * @param keyName name of the secret
+     * @return secret for the given key and application as char[]
+     */
+    default char[] getSecret(String appName, String keygroupName, String keyName) {
+        return keyName.toCharArray();
```

Comment:
this should be set to getSecret(appName, null, keyName); to main backward compatibility and not break anyone before they get a chance to update their code.